<!-- https://answers.ros.org/question/221093/basic-diff_drive-question/ -->
<!-- https://github.com/ros-controls/ros_controllers/tree/kinetic-devel/diff_drive_controller/test -->


<!-- https://github.com/eborghi10/my_ROS_mobile_robot -->

<launch>


<!--    
    <node name="arduino_node" pkg="rosserial_python" type="serial_node.py"/>
--> 
   <!-- todo: add /dev/ttyUSB0 option -->

    <group ns="zoef">
        <rosparam file="$(find zoef_ros_package)/config/zoef_base_config.yaml"/>
    </group>

    <node name="my_robot_base_node" 
          pkg="zoef_ros_package" 
          type="zoef_ros_package_node" output="screen"/>

    <node name="zoef_pymata"
          pkg="zoef_ros_package"
          type="pymata.py" output="screen"/>
          
    <node name="my_robot_camera"
          pkg="zoef_ros_package"
          type="camera.py" output="screen"/>

    <node name="zoef_navigation"
          pkg="zoef_ros_package"
          type="navigation.py" output="screen"/>
 
    <node name="zoef_service_api"
          pkg="zoef_ros_package"
          type="service_api.py" output="screen"/>



    <!-- Load controller config -->
    <rosparam command="load" 
              file="$(find zoef_ros_package)/config/my_robot_common.yaml"/>

    <node name="controller_spawner" 
          pkg="controller_manager" 
          type="spawner" 
          output="screen" 
          args="mobile_base_controller"/> 

    <include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />

<!--
  <node name="educational" pkg="educational_topics" type="turn.py" />
  <node name="educational2" pkg="educational_topics" type="move.py" />

  <node name="webcam" pkg="usb_cam" type="usb_cam_node">
     <param name="pixel_format" value="yuyv"/>
  </node>

  <node name="web_video_server" pkg="web_video_server" type="web_video_server"/>

  <node name="cam_mask" pkg="educational_topics" type="camera.py" />
    -->      
</launch>
